soc/intel/intel_adsp/ace/power_down.S

```diff
@@ -95,44 +85,25 @@ power_down:
 	l32i pfl_reg, p_ipc_regs, 0
 #endif
 
-	beqz pu32_hpsram_mask, _PD_DISABLE_LPSRAM
-	movi pu32_hpsram_mask, hpsram_mask
-
 _PD_DISABLE_LPSRAM:
 /**
  * effectively executes:
- * if (b_enable_lpsram) {
+ * if (b_disable_lpsram) {
  *     ace_lpsram_power_down_entire();
  * }
  */
-	beqz b_enable_lpsram, _PD_DISABLE_HPSRAM
+	beqz b_disable_lpsram, _PD_DISABLE_HPSRAM
 	m_ace_lpsram_power_down_entire temp_reg0, temp_reg1, temp_reg2, temp_reg3
 
 _PD_DISABLE_HPSRAM:
-	/* if value pu32_hpsram_mask = 0 - do not disable hpsram. */
-	beqz pu32_hpsram_mask, _PD_SEND_IPC
-	/**
-	 * effectively executes:
-	 * for (size_t seg_index = (MAX_MEMORY_SEGMENTS - 1); seg_index >= 0;
-	 * --seg_index) {
-	 * ace_hpsram_power_change(seg_index, mask[seg_index]);
-	 * }
-	 * where mask is given in pu32_hpsram_mask register
-	 */
-
-	.set seg_index, MAX_MEMORY_SEGMENTS - 1
-	.rept MAX_MEMORY_SEGMENTS
-		l32i temp_reg0, pu32_hpsram_mask, 4 * seg_index
-		m_ace_hpsram_power_change\
-			/*segment_index=*/	seg_index,\
-			/*mask=*/	temp_reg0,\
-			temp_reg1,\
-			temp_reg2,\
-			temp_reg3,\
-			temp_reg4,\
-			temp_reg5
-		.set seg_index, seg_index - 1
-	.endr
+/**
+ * effectively executes:
+ * if (b_disable_hpsram) {
+ *     ace_hpsram_power_down_entire();
+ * }
+ */
```

Comment:
also for a follow-up PR, maybe add a comment, saying that LPSRAM isn't used to run code, so we aren't running there now